### PR TITLE
AccountState: Avoid ConnectionCheck if ETag job was just done.

### PR DESCRIPTION
**Pull request by dragotin**
This patch lets a successful etag job check mark a timestamp.
If next time a connection check is requested, it is checked if
the last ETag happened within the last 30 seconds and if so the
connection check can be checked.

This way we avoid half of the PROPFINDs if all goes well.

@ckamm @ogoffart @guruz please check.

### CODE REVIEW

**Comment by guruz** [comment]
(haven't tested this theory)

With multiple sync folders/accounts the ETag checks will be done sequentially (at least that's how it was before multi-account)

Having a check with <= 30 sec will therefore mostly be false since those checks were sequential and could be longer than 30 sec ago

**Comment by guruz** [comment]
Please mention the related issue ID in the commit message

https://github.com/owncloud/client/issues/3677

**Comment by guruz** [comment]
Ah, and the 30 sec should be instead related to remotePollInterval ? (Plus the comment above I mentioned with regards to multiple accounts/folders)

**Comment by dragotin** [comment]
@guruz yes, exactly my thought, it should be the same as remotePollInterval. Will follow up.

**Comment by ogoffart** [comment]
I think the right fix is not to do the connection check chen the account is online.   The etag check will make the connection drop.

File: src/gui/accountstate.h [hunk 1]
```diff
@@ -95,6 +95,14 @@ class AccountState : public QObject {
      */
     QString shortDisplayNameForSettings(int width = 0) const;
 
+    /** Mark the timestamp when the last successful ETag check happened for
+     *  this account.
+     *  The checkConnectivity() method uses the timestamp to save a call to
+     *  the server to validate the connection if the last successful etag job
+     *  is not so lang away.
```
Comment (phil-davis) [inline src/gui/accountstate.h]: "was not too far in the past." ?

**Comment by guruz** [comment]
btw, we also must not break
```
void AccountState::checkConnectivity(CredentialFetchMode credentialsFetchMode)
...
#ifdef Q_OS_WIN
        // There seems to be a bug in Qt on Windows where QNAM sometimes stops
        // working correctly after the computer woke up from sleep. See #2895 #2899
        // and #2973.
        // As an attempted workaround, reset the QNAM regularly if the account is
        // disconnected.
        account()->resetNetworkAccessManager();
```

**Comment by guruz** [comment]
@dragotin You had asked me for an explanation on https://github.com/owncloud/client/pull/3964#issuecomment-149203019

What I meant is that ```FolderMan::slotEtagPollTimerTimeout()``` handles the ```ETagJob``` for all accounts and it only schedules one job at the same time. So if you have two accounts then your 30 sec check would not work.

I'll have a look at this and possibly fix this here.

**Comment by guruz** [comment]
Cherry picked into 2.1

**Pull request closed without merging**